### PR TITLE
Added filters to change the max and min supported WP versions

### PR DESCRIPTION
**Pull request by ginsterbusch**
I've added filter hooks to change the max and min supported WP versions in the preflight check, so that migration tests with older or current WP versions (eg. nightly builds) can be done without having to manually edit the plugin source code.

### CODE REVIEW

**Comment by nylen** [comment]
Hi @ginsterbusch, here are the changes I would recommend to this PR:

- Use a single filter `classicpress_ignore_wp_version`.  Elsewhere in the plugin, all function names are prefixed with `classicpress_`.  A couple of reasons for using a single filter:  I think it makes more sense to just bypass the version checking logic entirely, and this simplifies the usage of the filter (see example code below).

- Add a note to the readme about the new filters, as well as a code comment immediately above the filters.  It could say something like this:

> If you would like to migrate from a different version of WordPress AT YOUR OWN RISK, you can use the following code in your theme or a mu-plugin to enable the migration:
>
> `add_filter( 'classicpress_ignore_wp_version', '__return_true' );`

- If the version check is overridden with this filter, show a special message in the dashboard with the yellow flag icon, reiterating that the migration is at your own risk.

File: lib/admin-page.php [hunk 1]
```diff
@@ -304,33 +304,59 @@ function classicpress_check_can_migrate() {
...
 		$preflight_checks['wp_version'] = true;
-		echo "<tr>\n<td>$icon_preflight_pass</td>\n<td>\n";
+		
+	} else { // Run the check
+		global $wp_version;
+		
+		$wp_version_min = '4.9.0';
+		$wp_version_max = '4.9.8';
+		
+		// Default values - skips one else
```
Comment (nylen) [inline lib/admin-page.php]: When you see a code comment like this, you know things are starting to get a bit convoluted :)  Here is the way I think we should structure this code instead:

`if ( version check failed ) {
    if ( apply_filters( 'classicpress_ignore_wp_version', false ) ) {
        // Show warning but allow migration to proceed
    } else {
        // Block migration, show error message
    }
} else {
    // Everything OK, and we don't need to worry about the value of the new filter
}
```

File: lib/admin-page.php [hunk 2]
```diff
@@ -304,33 +304,59 @@ function classicpress_check_can_migrate() {
...
 	}
+		
+	echo "<tr>\n<td>$preflight_check_wp_version_status</td>\n<td>\n";
+	
 	echo "<p>\n";
+	
+	// Warn the user thoroughly: From here there be dragons!
+	if( !empty( $disable_wp_version_check ) ) {
+		_e( 
+			'<strong>Please be aware:</strong> The preflight check for supported WordPress versions has been <strong>disabled</strong>. Without this, we cannot guarantee that the migration process is going to work, and may even leave your current installation partially broken. Proceed on your own risk!',
```
Comment (nylen) [inline lib/admin-page.php]: - Proceed **at** your own risk.
- Might be good to explain **why** this is risky, if there is room for it:  We haven't tested this and we don't know what's going to happen to your site.

File: lib/admin-page.php [hunk 3]
```diff
@@ -304,33 +304,59 @@ function classicpress_check_can_migrate() {
...
+	}
+	
 	printf( __(
 		/* translators: 1: minimum supported WordPress version, 2: maximum supported WordPress version */
 		'This plugin supports WordPress versions <strong>%1$s</strong> to <strong>%2$s</strong> (and some newer development versions).',
 		'switch-to-classicpress'
 	), $wp_version_min, $wp_version_max );
-	echo "<br>\n";
+	
+	echo "<br />\n";
```
Comment (nylen) [inline lib/admin-page.php]: I don't have a problem with this change. ([It's a mess, but either way is valid HTML5](https://stackoverflow.com/questions/1946426/html-5-is-it-br-br-or-br).)

However, it would be good to make this change in a separate PR and update all of these tags at once.

Comment (ginsterbusch) [reply lib/admin-page.php]: Personally I'd totally rewrite the plugin with proper OOP, seperate templates, CSS et all, but I do understand the need for a basic, plain code setup.

Comment (nylen) [reply lib/admin-page.php]: That has been proposed before (cc @glagonikas).  I think it is not urgent but it is probably worth doing in a separate PR.  I'd use a class for each check, and methods with standardized return values to indicate their status.

I also expect we will want to re-use this code structure for upgrades between ClassicPress versions, so it would be nice to keep it more maintainable.

**Comment by nylen** [comment]
Updated, and merging along with the changes to support migrating from WP 5.0.
